modin/pandas/test/dataframe/test_map_metadata.py

```diff
@@ -557,6 +558,57 @@ def test_astype_category_large():
     assert modin_result.dtypes.equals(pandas_result.dtypes)
 
 
+@pytest.mark.parametrize("infer_objects", (True, False), ids=("infer_objects", ""))
+@pytest.mark.parametrize("convert_string", (True, False), ids=("convert_string", ""))
+@pytest.mark.parametrize("convert_integer", (True, False), ids=("convert_integer", ""))
+@pytest.mark.parametrize("convert_boolean", (True, False), ids=("convert_boolean", ""))
+@pytest.mark.parametrize(
+    "convert_floating", (True, False), ids=("convert_floating", "")
+)
+def test_convert_dtypes(
+    infer_objects, convert_string, convert_integer, convert_boolean, convert_floating
+):
+    # Sanity check, copied from pandas documentation:
+    # https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.convert_dtypes.html
+    data = {
+        "a": pd.Series([1, 2, 3], dtype=np.dtype("int32")),
+        "b": pd.Series(["x", "y", "z"], dtype=np.dtype("O")),
+        "c": pd.Series([True, False, np.nan], dtype=np.dtype("O")),
+        "d": pd.Series(["h", "i", np.nan], dtype=np.dtype("O")),
+        "e": pd.Series([10, np.nan, 20], dtype=np.dtype("float")),
+        "f": pd.Series([np.nan, 100.5, 200], dtype=np.dtype("float")),
+    }
+    kwargs = {
+        "infer_objects": infer_objects,
+        "convert_string": convert_string,
+        "convert_integer": convert_integer,
+        "convert_boolean": convert_boolean,
+        "convert_floating": convert_floating,
+    }
+    modin_df = pd.DataFrame(data)
+    pandas_df = pandas.DataFrame(data)
+    modin_result = modin_df.convert_dtypes(**kwargs)
+    pandas_result = pandas_df.convert_dtypes(**kwargs)
+    assert modin_result.dtypes.equals(pandas_result.dtypes)
+
+
+def test_convert_dtypes_row_partition():
+    # Tests `convert_dtypes` when values in the same column may come from different
+    # row partitions.
+    part_col_size = MinPartitionSize.get()
```

Comment:
`part_col_size` isn't useful now that we're forcing multiple partitions with `concat`. We can initialize each dataframe with a single element.

Reply:
Since as you said above, having multiple partitions isn't guaranteed with `concat`, would it make more sense to directly construct a single dataframe with something like `pd.DataFrame((["a"] * part_col_size) + ([1] * part_col_size))`?

Reply:
We should never treat multiple partitions as guaranteed, though generally we assume in our unit tests that constructing a frame with more than `MinPartitionSize` elements on an axis forces multiple partitions on that axis. I think the way you have it is fine for now. Let's just add the assert I mentioned in my other comment.